components/select/Select.razor.cs

```diff
@@ -84,33 +86,29 @@ public IEnumerable<TItem> DataSource
                     SelectedOptionItems.Clear();
 
                     Value = default;
```

Comment:
I don't think we should change the value inside DataSource property. As I mentioned, the order to set `Value` and `DataSource` depends on user hand-writing. And this is actually unnecessary.

Reply:
If our proposal of removing value changes on `DataSource` changes gets accepted, then this will most likely be removed from the setter.